MainWindow.cpp

```diff
@@ -73,15 +77,44 @@ void diagnosticHandler(QtMsgType type, const QMessageLogContext &context, const
 
   std::cerr << msgFormatted.toStdString() << std::endl;
 
-  if (diagnosticTextEdit) {
+  if (diagnosticTextEdit != nullptr) {
     diagnosticTextEdit->append(msgFormatted);
   } else {
     // this should never happen!
     std::cerr << "Critical: Diagnostics text control does not exist!" << std::endl;
   }
 }
 
-MainWindow::MainWindow(QWidget *parent) : QMainWindow(parent), _ui(new Ui::MainWindow) {
+QFileInfo MainWindow::getEnigmaRoot() {
+  QFileInfo EnigmaRoot;
+  foreach (auto path, EnigmaSearchPaths) {
+    const QDir dir(path);
+    QDir::Filters filters = QDir::Filter::AllEntries;
+    auto entryList = dir.entryInfoList(QStringList({"ENIGMAsystem"}), filters, QDir::SortFlag::NoSort);
+    if (!entryList.empty()) {
+      EnigmaRoot = entryList.first();
+        break;
+    }
+  }
+  
+  return EnigmaRoot;
+}
+
+MainWindow::MainWindow(QWidget *parent) : QMainWindow(parent), _ui(new Ui::MainWindow), _event_data(nullptr), egm(nullptr) {
+
+  if (!EnigmaRoot.filePath().isEmpty()) {
+    _event_data = std::make_unique<EventData>(ParseEventFile((EnigmaRoot.absolutePath() + "/events.ey").toStdString()));

```

Comment:
Ask @JoshDreamland why `ParseEventFile` doesn't just return a `unique_ptr` for me.

Reply:
EventData should be an object not a pointer but because it doesn't have a default constructor I need to conditionally initialize it with different constructors by making it a pointer.